Tests/scripts/utils/compare_staging_graph.py

```diff
@@ -0,0 +1,189 @@
+# function to compare two zip files
+
+import argparse
+import filecmp
+import json
+import os
+from pathlib import Path
+import shutil
+from zipfile import ZipFile
+
+import difflib
+from contextlib import redirect_stdout
+import dictdiffer
+
+from ruamel.yaml import YAML
+from slack_sdk import WebClient
+
+yaml = YAML()
+
+
+def sort_dict(dct: dict):
+    for k, v in dct.items():
+        if isinstance(v, dict):
+            sort_dict(v)
+        if isinstance(v, list):
+            try:
+                v.sort()
+            except TypeError:
+                v.sort(key=lambda item: item.get("name"))
+
+
+def compare_indexes(index_id_set_path: Path, index_graph_path: Path, output_path: Path) -> bool:
+    with index_id_set_path.open() as f1, index_graph_path.open() as f2:
+        index_id_set = json.load(f1)
+        index_graph = json.load(f2)
+        index_id_set.pop("modified", None)
+        index_graph.pop("modified", None)
+        sort_dict(index_id_set)
+        sort_dict(index_graph)
+        diff_list = list(dictdiffer.diff(index_id_set, index_graph))
+    if diff_list:
+        with (output_path / "index-diff.json").open("w") as output:
+            json.dump(diff_list, output, indent=4)
+        shutil.copyfile(index_id_set_path, output_path / "index-id_set.json")
+        shutil.copyfile(index_graph_path, output_path / "index-graph.json")
+        return True
+    return False
+
+
+def compare_dirs(dir1: str, dir2: str, output_path: Path) -> list[str]:
+    dir_compare = filecmp.dircmp(dir1, dir2)
+    full_report_path = output_path / "full_report.log"
+    with open(full_report_path, "w") as f:
+        with redirect_stdout(f):
+            dir_compare.report_full_closure()
+    diff_files: list[str] = []
+    compare_files(dir_compare, dir1, dir2, output_path, diff_files)
+    return diff_files
+
+
+def compare_zips(zip1: Path, zip2: Path, output_path: Path) -> list[str]:
+    """Compare two zip files content"""
+    # extract zip files
+    output_path.mkdir(parents=True, exist_ok=True)
+    zip1_files = str(output_path / "id_set_tmp")
+    zip2_files = str(output_path / "graph_tmp")
+    with ZipFile(zip1, "r") as zip1_content, ZipFile(zip2, "r") as zip2_content:
+        # get the list of files in the zip files
+        zip1_content.extractall(path=zip1_files)
+        zip2_content.extractall(path=zip2_files)
+
+    # compare the directories
+    return compare_dirs(zip1_files, zip2_files, output_path)
+
+
+def compare_files(
```

Comment:
how do we manage files to skip comparison?

Reply:
added global SKIPPED_FILES